packages/bun-types/globals.d.ts

```diff
@@ -1286,6 +1286,88 @@ interface ImportMeta {
    */
   main: boolean;
 
+  /**
+   * Import multiple modules using glob patterns.
+   *
+   * @param pattern - A glob pattern or array of glob patterns to match files
+   * @param options - Options for how imports are handled
+   * @returns An object mapping file paths to import functions or modules
+   *
+   * @example
+   * const modules = import.meta.glob('./modules/*.js')
+   * const module = await modules['./modules/foo.js']()
+   *
+   * // code produced by bun
+   * const modules = {
+   *   './modules/foo.js': () => import('./modules/foo.js'),
+   *   './modules/bar.js': () => import('./modules/bar.js'),
+   * }
+   * const module = await modules['./modules/foo.js']()
+   */
+  glob<Eager extends boolean = false, TModule = unknown>(
+    pattern: string | string[],
+    options?: {
+      // todo:
+      // /**
+      //  * If true, imports all modules eagerly (synchronously).
+      //  * If false (default), returns functions that import modules lazily.
+      //  */
+      // eager?: Eager;
+      eager?: false;
+      /**
+       * Specify a named export to import from matched modules.
+       * If not specified, imports the entire module.
+       *
+       * @example
+       * const modules = import.meta.glob('./dir/*.js', { import: 'setup' })
+       *
+       * // code produced by bun
+       * const modules = {
+       *   './dir/bar.js': () => import('./dir/bar.js').then((m) => m.setup),
+       *   './dir/foo.js': () => import('./dir/foo.js').then((m) => m.setup),
+       * }
+       */
+      import?: string;
+      /**
+       * Add a query string to the end of the "generated" dynamic import.
+       *
+       * @example
+       * const modules = import.meta.glob('./assets/*.txt', { query: '?something' })
+       *
+       * // code produced by bun
+       * const modules = {
+       *   './assets/file.txt': () => import('./assets/file.txt?something'),
+       * }
+       */
+      query?: string;
```

Comment:
Realistically there isn't too much point adding this as vite has `?raw` type thing, but someone may have a usecase. Also if they were needing to use it to clear cache or with `build --no-bundle` and using a different runtime which needs.